go/vt/discovery/tablet_picker.go

```diff
@@ -92,16 +159,108 @@ func NewTabletPicker(ts *topo.Server, cells []string, keyspace, shard, tabletTyp
 		return nil, vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION,
 			fmt.Sprintf("Missing required field(s) for tablet picker: %s", strings.Join(missingFields, ", ")))
 	}
+
+	// Resolve tablet picker options
+	cellPref := parseTabletPickerCellPreferenceString(options.CellPref)
+
+	// For backward compatibility only parse the options for tablet ordering
+	// if the in_order hint wasn't already specified. Otherwise it could be overridden.
+	// We can remove this check once the in_order hint is deprecated.
+	if !inOrder {
+		order := parseTabletPickerTabletOrderString(options.TabletOrder)
+		switch order {
+		case TabletPickerTabletOrder_Any:
+			inOrder = false
+		case TabletPickerTabletOrder_InOrder:
+			inOrder = true
+		}
```

Comment:
by flag's default do you mean the default returned from the proto definition if the value is blank?

Reply:
I might be misunderstanding your suggestion, but I think it's better for now to maybe have the default defined in the constructor of  the tablet picker, just because there are various places where the options have to be passed in explicitly and it's easier to pass in a blank and know that the constructor will choose the right default for us: https://github.com/vitessio/vitess/pull/12282/files#diff-930d6d6efe03db7434759f4cb3af5b8a7783ad68e9c00a21f5bb21ac7c76359aR252

^^ the above does not have a flag yet